### PR TITLE
Issue #1028: Make the layout list more useful.

### PR DESCRIPTION
**Pull request by jenlampton**
PR for https://github.com/backdrop/backdrop-issues/issues/1028

### CODE REVIEW

File: core/modules/layout/layout.theme.inc [hunk 1]
```diff
@@ -46,8 +46,44 @@ function theme_layout_reorder_layouts($variables) {
  */
 function theme_layout_info($variables) {
   $layout = $variables['layout'];
+  $info = layout_get_layout_info($layout->layout_template);
```
Comment (quicksketch) [inline core/modules/layout/layout.theme.inc]: `layout_get_layout_info()` is deprecated, use `layout_get_layout_template_info()`.

File: core/modules/layout/css/layout.admin.css [hunk 2]
```diff
@@ -12,6 +12,41 @@
...
+  }
+}
+
+#layout-list .layout-conditions-list ul {
+  margin-left: 0;
+}
+
+#layout-list .item-list ul li {
+  list-style-type: square;
+}
```
Comment (quicksketch) [inline core/modules/layout/css/layout.admin.css]: Per @klonos's feedback in the previous PR, let's remove this square list type.

File: core/modules/layout/tests/layout.test [hunk 3]
```diff
@@ -61,7 +94,7 @@ class LayoutInterfaceTest extends BackdropWebTestCase {
     $edit = array(
       'title' => $layout_title,
       'name' => $layout_name,
-      'layout_template' => 'two_column',
+      'layout' => 'two_column',
```
Comment (quicksketch) [inline core/modules/layout/tests/layout.test]: These are regressions.

File: core/modules/layout/layout.theme.inc (hunk 1 above)
Comment (jenlampton) [inline core/modules/layout/layout.theme.inc]: fixed.

File: core/modules/layout/css/layout.admin.css (hunk 2 above)
Comment (jenlampton) [inline core/modules/layout/css/layout.admin.css]: removed.

File: core/modules/layout/tests/layout.test (hunk 3 above)
Comment (jenlampton) [inline core/modules/layout/tests/layout.test]: unregressed

File: core/modules/layout/tests/layout.test [hunk 4]
```diff
@@ -1115,7 +1148,7 @@ class LayoutInterfaceTest extends BackdropWebTestCase {
 
     // Change the layout to the test layout, which has identical regions.
     $edit = array(
-      'layout_template' => 'layout_test_layout',
+      'layout' => 'layout_test_layout',
```
Comment (quicksketch) [inline core/modules/layout/tests/layout.test]: Still have some terminology regressions here (and below).

**Comment by jenlampton** [comment]
Pushed an updated PR.

File: core/modules/layout/layout.theme.inc [hunk 5]
```diff
@@ -46,8 +46,44 @@ function theme_layout_reorder_layouts($variables) {
...
+  $output .= $preview;
+  $output .= '<div class="layout-detail">';
+  $output .= '  <div class="layout-title">' . check_plain($layout->title) . '</div>';
+  $output .= '  <div class="layout-template description">' . t('<span class="priority-low">Template: </span>!title', array('!title' => $template)) . '</div>';
+  $output .= '</div></div>';
+
+  return $output;
+}
+
+function theme_layout_condition_info($variables) {
```
Comment (quicksketch) [inline core/modules/layout/layout.theme.inc:78]: This function could use a docblock.

File: core/modules/layout/layout.theme.inc [hunk 6]
```diff
@@ -46,8 +46,44 @@ function theme_layout_reorder_layouts($variables) {
...
+  $layout = $variables['layout'];
+  // Build a list of visibility conditions on this layout
+  $conditions = array();
+  foreach ($layout->conditions as $access_key => $layout_access) {
+    $conditions[] = $layout_access->summary();
+  }
+  $list = '';
```
Comment (quicksketch) [inline core/modules/layout/layout.theme.inc]: Unused variable `$list`.

File: core/modules/layout/layout.theme.inc (hunk 5 above)
Comment (jenlampton) [inline core/modules/layout/layout.theme.inc:78]: added.

File: core/modules/layout/layout.theme.inc (hunk 6 above)
Comment (jenlampton) [inline core/modules/layout/layout.theme.inc]: removed.

File: core/modules/layout/layout.admin.inc [hunk 7]
```diff
@@ -29,11 +29,15 @@ function layout_list_page() {
     $row = array();
     // Default layouts have a null path.
     if ($path === '') {
-      $row[] = array('data' => t('Default layouts (used on all paths without a specific layout)'), 'colspan' => 2);
+      $row[] = array('data' => t('Default layouts (used on all paths without a specific layout)'));
+      $row[] = '';
+      $row[] = '';
```
Comment (klonos) [inline core/modules/layout/layout.admin.inc:34]: Perhaps silly question: what are these empty rows used for?

File: core/modules/layout/layout.admin.inc [hunk 8]
```diff
@@ -29,11 +29,15 @@ function layout_list_page() {
...
     }
     else {
       $path_link = strpos($path, '%') === FALSE ? l($path, $path) : check_plain($path);
-      $row[] = array('data' => $path_link, 'colspan' => 2);
+      $row[] = array('data' => $path_link);
+      $row[] = '';
+      $row[] = '';
```
Comment (klonos) [inline core/modules/layout/layout.admin.inc:40]: ...same question as above.

File: core/modules/layout/layout.admin.inc (hunk 7 above)
Comment (docwilmot) [inline core/modules/layout/layout.admin.inc:34]: Its to mitigate the problem you found at https://github.com/backdrop/backdrop/pull/1341#issuecomment-215006463 with the responsiveness of the title rows. I found that the mechanism for hiding columns based on priority tags in the header (eg `priority-low`) was negated by having `colspan` set on the row, and there is no way I could find to resize those rows with CSS as there is no CSS version of `colspan`. So instead, dont have one row that spans 3 columns, just have 3 colums, two of which are empty. The alternative is to resize with JS, but I thought this was simpler.

**Comment by quicksketch** [comment]
Tests are failing in the current PR still. Seems to be related to the "layout" => "layout_template" rename.

**Comment by klonos** [comment]
@docwilmot thanx for the explanation. That makes perfect sense then. Can we add some comment there though so that the next person that sees it doesn't wonder?

**Comment by klonos** [comment]
...never mind the details. Just sort the important things and I'll be filing task-issues for small things that can be done after 1.4.0 :wink:

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/1389, which fixes the failing test.

**Pull request closed without merging**